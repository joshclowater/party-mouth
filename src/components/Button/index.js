import styled from 'styled-components';

export default styled.button`
  border: 1px solid transparent;
  padding: 1vh 1.8vh;
  font-size: 2.5vh;
  border-radius: 0.6vh;
  cursor: ${props => (props.disabled ? 'not-allowed' : 'pointer')};

  color: #fff;
  background-color: #007bff;
  border-color: #007bff;

  ${props =>
    props.disabled
      ? 'opacity: 0.65;'
      : `:hover {
      background-color: #0069d9;
      border-color: #0062cc;
    }
    transition: color .15s ease-in-out,background-color .15s ease-in-out,border-color .15s ease-in-out,box-shadow .15s ease-in-out;`};
`;
