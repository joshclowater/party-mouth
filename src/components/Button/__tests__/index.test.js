import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';
import Button from '../';

it('Renders correctly', () => {
  expect(shallow(<Button />)).toMatchSnapshot();
});

describe('disabled', () => {
  it('Renders correctly', () => {
    expect(shallow(<Button disabled />)).toMatchSnapshot();
  });
});
