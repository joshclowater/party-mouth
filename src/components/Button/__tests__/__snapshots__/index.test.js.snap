// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
.c0 {
  border: 1px solid transparent;
  padding: 1vh 1.8vh;
  font-size: 2.5vh;
  border-radius: 0.6vh;
  cursor: pointer;
  color: #fff;
  background-color: #007bff;
  border-color: #007bff;
  -webkit-transition: color .15s ease-in-out,background-color .15s ease-in-out,border-color .15s ease-in-out,box-shadow .15s ease-in-out;
  transition: color .15s ease-in-out,background-color .15s ease-in-out,border-color .15s ease-in-out,box-shadow .15s ease-in-out;
}

.c0:hover {
  background-color: #0069d9;
  border-color: #0062cc;
}

<button
  className="c0"
/>
`;

exports[`disabled Renders correctly 1`] = `
.c0 {
  border: 1px solid transparent;
  padding: 1vh 1.8vh;
  font-size: 2.5vh;
  border-radius: 0.6vh;
  cursor: not-allowed;
  color: #fff;
  background-color: #007bff;
  border-color: #007bff;
  opacity: 0.65;
}

<button
  className="c0"
  disabled={true}
/>
`;
